src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -1034,6 +1051,91 @@ private void rejectNorthernIrelandHouseholdToCE(
     }
   }
 
+  private void rejectIfForCrownDependency(String postcode) throws CTPException {
+    String postcodeArea = postcode.substring(0, 2).toUpperCase();
+
+    switch (postcodeArea) {
+      case "GY":
+      case "JE":
+        log.with(postcode).info("Rejecting request as postcode is for a channel island address");
+        throw new CTPException(
+            Fault.BAD_REQUEST, "Channel Island addresses are not valid for Census");
+      case "IM":
+        log.with(postcode).info("Rejecting request as postcode is for an Isle of Man address");
+        throw new CTPException(Fault.BAD_REQUEST, "Isle of Man addresses are not valid for Census");
+      default: // to keep checkstyle happy
+    }
+  }
+
+  private uk.gov.ons.ctp.integration.contactcentresvc.representation.Region determineActualRegion(
+      NewCaseRequestDTO caseRequestDTO) throws CTPException {
+    uk.gov.ons.ctp.integration.contactcentresvc.representation.Region requestedRegion =
+        caseRequestDTO.getRegion();
+    uk.gov.ons.ctp.integration.contactcentresvc.representation.Region actualRegion = null;
+
+    String postcode = caseRequestDTO.getPostcode();
+    String postcodeArea = postcode.substring(0, 2).toUpperCase();
+
+    if (postcodeArea.equals("BT")) {
+      log.with(postcode).debug("Forcing region to Northern Ireland");
+      actualRegion = uk.gov.ons.ctp.integration.contactcentresvc.representation.Region.N;
+
+    } else {
+      // Get ready to call AI to find the region for the specified postcode
+      MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
+      queryParams.add("offset", "0");
+      queryParams.add("limit", "1");
+      queryParams.add("includeauxiliarysearch", "true");
+      addEpoch(queryParams);
+
+      // Ask Address Index to do postcode search
+      AddressIndexSearchResultsDTO addressIndexResponse = null;
+      try {
+        String path = appConfig.getAddressIndexSettings().getPostcodeLookupPath();
+        addressIndexResponse =
+            addressIndexClient.getResource(
+                path, AddressIndexSearchResultsDTO.class, null, queryParams, postcode);
+      } catch (ResponseStatusException e) {
+        // Something went wrong calling AI.
+        // Never mind, we'll still be able to use the Serco supplied region
+        log.with(postcode).warn("Failed to call AI to resolve region");
+      }
+
+      if (addressIndexResponse != null) {
+        ArrayList<AddressIndexAddressDTO> addresses =
+            addressIndexResponse.getResponse().getAddresses();
+        if (!addresses.isEmpty()) {
+          // Found an address. Fail if Scottish otherwise use its region
+          String countryCode = addresses.get(0).getCensus().getCountryCode();
+          if (countryCode.equals("S")) {
```

Comment:
is it ever possible that countryCode could come back null from AIMS.  Should we guard against it ?

Reply:
Code updated to prevent NPE if/when this happens.
Unit test added for this case